server/sonar-server/src/main/java/org/sonar/server/computation/period/Period.java

```diff
@@ -22,19 +22,22 @@
 
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
-import org.apache.commons.lang.builder.ReflectionToStringBuilder;
+import javax.annotation.concurrent.Immutable;
 
+@Immutable
 public class Period {
```

Comment:
a snapshot is something the period is associated to, so having a object for that is relevant
but it saves only one property... for now it may not be worth it